lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -207,113 +207,138 @@ module.exports = {
-          'describeSubscriptionFilters',
-          { logGroupName },
-          { useCache: true }
-        )
-        .then((response) => {
-          const subscriptionFilter = response.subscriptionFilters[0];
+    const logGroupName = params.logGroupName;
+    const cloudwatchLogEvents = params.cloudwatchLogEvents;
+    const CLOUDWATCHLOG_LOG_GROUP_EVENT_PER_FUNCTION_LIMIT = 2;
+
+    const response = await this.provider
+      .request(
+        'CloudWatchLogs',
+        'describeSubscriptionFilters',
+        { logGroupName },
+        { useCache: true }
+      )
+      .catch(() => ({ subscriptionFilters: [] }));
+    if (response.subscriptionFilters.length === 0) {
+      return false;
+    }
 
-          // log group doesn't have any subscription filters currently
-          if (!subscriptionFilter) {
-            return false;
-          }
+    const oldSubscriptionFilters = response.subscriptionFilters.map((subscriptionFilter) => ({
+      ...subscriptionFilter,
+      logicalId: this.getLogicalIdFromFilterName(subscriptionFilter.filterName),
+    }));
 
-          const filterName = subscriptionFilter.filterName;
+    const newSubscriptionFilters = cloudwatchLogEvents.map((cloudwatchLogEvent) => {
+      const destinationArn = `arn:${partition}:lambda:${region}:${accountId}:function:${cloudwatchLogEvent.FunctionName}`;
+      const logicalId = this.provider.naming.getCloudWatchLogLogicalId(
+        cloudwatchLogEvent.functionName,
+        cloudwatchLogEvent.logSubscriptionSerialNumber
+      );
 
-          const oldDestinationArn = subscriptionFilter.destinationArn;
-          const newDestinationArn = `arn:${partition}:lambda:${region}:${accountId}:function:${functionObj.name}`;
-          const oldLogicalId = this.getLogicalIdFromFilterName(filterName);
-          const newLogicalId = this.provider.naming.getCloudWatchLogLogicalId(
-            functionName,
-            logSubscriptionSerialNumber
-          );
+      return {
+        destinationArn,
+        logicalId,
+      };
+    });
 
-          // everything is fine, just return
-          if (oldDestinationArn === newDestinationArn && oldLogicalId === newLogicalId) {
-            return false;
-          }
+    const sameDestinationArn = (sf1, sf2) => sf1.destinationArn === sf2.destinationArn;
+    const sameLogicalId = (sf1, sf2) => sf1.logicalId === sf2.logicalId;
+    const subscriptionFilterComparator = (sf1, sf2) =>
+      sameDestinationArn(sf1, sf2) && sameLogicalId(sf1, sf2);
 
-          /*
-            If the destinations functions' ARNs doesn't match, we need to delete the current
-            subscription filter to prevent the resource limit exceeded error to happen
-          */
-          return this.provider.request('CloudWatchLogs', 'deleteSubscriptionFilter', {
-            logGroupName,
-            filterName,
-          });
+    const matchedOldSubscriptionFilters = oldSubscriptionFilters.filter((oldSubscriptionFilter) => {
```

Comment:
Let's maybe put a comment above this variable that explains in more details what does it mean - at first sight it's not obvious what `notMatched` or `matched` means